couchbase/tests/test_couchbase.py

```diff
@@ -194,7 +187,47 @@ def _assert_bucket_metrics(aggregator, tags, device=None):
 def _assert_stats(aggregator, node_tags, device=None):
     for mname in NODE_STATS:
         aggregator.assert_metric('couchbase.by_node.{}'.format(mname), tags=node_tags, count=1, device=device)
-
     # Assert 'couchbase.' metrics
     for mname in TOTAL_STATS:
         aggregator.assert_metric('couchbase.{}'.format(mname), tags=CHECK_TAGS, count=1)
+
+
+@pytest.mark.skipif(COUCHBASE_MAJOR_VERSION != 7, reason='Index metrics are only available for Couchbase 7+')
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_index_stats_metrics(aggregator, dd_run_check, instance_index_stats, couchbase_container_ip):
+    """
+    Test Index Statistics metrics (prefixed "couchbase.index." and "couchbase.indexer.")
+    """
+    couchbase = Couchbase('couchbase', {}, [instance_index_stats])
+    dd_run_check(couchbase)
+    for mname in INDEX_STATS_INDEXER_METRICS:
+        aggregator.assert_metric(mname, tags=CHECK_TAGS)
+
+    for mname in INDEX_STATS_INDEX_METRICS:
+        aggregator.assert_metric(mname, tags=INDEX_STATS_TAGS)
+    aggregator.assert_service_check(INDEX_STATS_SERVICE_CHECK_NAME, status=Couchbase.OK, tags=CHECK_TAGS)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_metrics(aggregator, instance, couchbase_container_ip):
+    """
+    Test couchbase metrics not including 'couchbase.query.'
+    """
+    # Few metrics are only available after some time post launch. Sleep to ensure they're present before we validate
+    time.sleep(15)
+    couchbase = Couchbase('couchbase', {}, instances=[instance])
+    couchbase.check(None)
```

Comment:
```suggestion
   dd_run_check(couchbase)
```
nit- I know you didn't write this code but since the test was updated let's use dd_run_check since it runs config validation

Reply:
Out of curiosity, what does that do exactly?

Reply:
The `integration.check()` method just runs the actual check, but  `dd_run_check(check)` runs the check and [check initializations](https://datadoghq.dev/integrations-core/base/basics/#check-initializations), which by default includes validating [configuration](https://datadoghq.dev/integrations-core/meta/config-models/). Running configuration validation can catch errors about our config specs in tests before they happen, such as when we incorrectly mark an option as `required`, when it should be `optional`